webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
 
-          boolean searchOk = ok;
+      ArrayList<String> arrayToDeClassify = new ArrayList<String>();
+      int begin = 0;
+      int end = 0;
+      end = listeSite.indexOf(',', begin);
+      String idSiteToDeClassify = null;
+      String pubId = null;
+      Collection<NodePK> listNodePk;
+      while (end != -1) {
+        pubId = listeSite.substring(begin, end); // pubId
 
-          SiteDetail descriptionSite2 =
-              new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, type, null,
-                  null, Integer.parseInt(etat), popup);
+        scc.removePublicationToFolder(pubId, id);
 
-          if (searchOk) {
+            /* isPublished dans un autre theme */
+        listNodePk = scc.getAllFatherPK(pubId);
+        if (listNodePk.size() == 0) {
+          PublicationDetail pubDetail = scc.getPublicationDetail(pubId);
+          idSiteToDeClassify = pubDetail.getVersion();
+          arrayToDeClassify.add(idSiteToDeClassify);
+        }
 
-            /* update description en BD */
-            scc.updateWebSite(descriptionSite2);
+        begin = end + 1;
+        end = listeSite.indexOf(',', begin);
+      }
 
-            if (listIcons.size() > 0) {
-              scc.associateIcons(id, listIcons);
-            }
-          } else {
-            request.setAttribute("SearchOK", Boolean.FALSE);
-            request.setAttribute("ListeIcones", listeIcones);
-          }
+      if (arrayToDeClassify.size() > 0) {
+        scc.dePublish(arrayToDeClassify); // set etat du site a 0
+      }
+      action = "Search";
+    }
 
-          descriptionSite2 = scc.getWebSite(id);
-          scc.setSessionSite(descriptionSite2);
-          request.setAttribute("Site", descriptionSite2);
-        } else if ("addFolder".equals(action)) {
-          // = null la premiere fois, puis = nom du repertoire courant
-          String name = request.getParameter("name");
+    if (id == null) {
+      id = "0";
+    }
 
-          // ADD FOLDER -------------------------------------------------------------
-          /* Creer le nouveau repertoire */
-          scc.createFolder(currentPath + "/" + name);
+    FolderDetail webSitesCurrentFolder = scc.getFolder(id);
+    scc.setSessionTopic(webSitesCurrentFolder);
+    request.setAttribute("CurrentFolder", webSitesCurrentFolder);
 
-          request.setAttribute("Site", scc.getSessionSite());
+    return "/webSites/jsp/organize.jsp?Action=" + action + "&Id=" + id + "&Path=" + path;
+  }
 
-        } else if ("renameFolder".equals(action)) {
-          // = null la premiere fois, puis = nom du repertoire courant
-          String name = request.getParameter("name");
-          // = changement de noms des fichiers et repertoires
-          String newName = request.getParameter("newName");
+  private String modifySiteDescription(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "userRole". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1172)